File: recipes/api/01_connect/connect.go
// connect.go: connect to MySQL server
package main

import (
    "database/sql"
    "fmt"
    "log"

    _ "github.com/go-sql-driver/mysql"
)

func main() {

    db, err := sql.Open("mysql", "cbuser:cbpass@tcp(127.0.0.1:3306)/cookbook")
    defer db.Close()

    if err != nil {
        log.Fatal(err)
    }

    fmt.Println("Connected!")
}

File: recipes/api/01_connect/connect.rb
#!/usr/bin/ruby
# connect.rb: connect to the MySQL server

require "dbi"

begin
  dsn = "DBI:Mysql:host=localhost;database=cookbook"
  dbh = DBI.connect(dsn, "cbuser", "cbpass")
  puts "Connected"
rescue => e
  puts "Cannot connect to server"
puts e.backtrace
  exit(1)
end
dbh.disconnect
puts "Disconnected"
